### PR TITLE
Added aseries support for error functions

### PR DESCRIPTION
**Pull request by 0sidharth**
#### References to other Issues or PRs

Continuing https://github.com/sympy/sympy/pull/17303

#### Brief description of what is fixed or changed
Added series support for error functions
```py
In [1]: Si(x).series(x, oo)
Out[1]: 
    ⎛  6    1    ⎛1        ⎞⎞          ⎛24   2         ⎛1        ⎞⎞       
    ⎜- ── + ─ + O⎜──; x → ∞⎟⎟⋅sin(x)   ⎜── - ── + 1 + O⎜──; x → ∞⎟⎟⋅cos(x)
    ⎜   3   x    ⎜ 5       ⎟⎟          ⎜ 4    2        ⎜ 5       ⎟⎟       
π   ⎝  x         ⎝x        ⎠⎠          ⎝x    x         ⎝x        ⎠⎠       
─ - ──────────────────────────────── - ───────────────────────────────────
2                  x                                    x                 

```
#### Release Notes

* functions
    * Added `aseries` expansion to error functions

### CODE REVIEW

**Comment by 0sidharth** [comment]
Current status:
```py
In [1]: erf(x).series(x, oo)
Out[1]: 
     ⎛1        ⎞
1 + O⎜──; x → ∞⎟
     ⎜ 6       ⎟
     ⎝x        ⎠
```

Reposting from gitter - 
When `_eval_nseries` calls `_eval_aseries` (L697 in function.py), adding a print statement says the asymptotic series being returned is
```
-(_x - _x**3/2 + 3*_x**5/4 + O(_x**6))*exp(-1/_x**2)/sqrt(pi) + 1
```
which is what I am expecting, but on returning this expression changes to
```
1 + O(x**(-6), (x, oo))
```
I think the answer sympy is finally returning is wrongly simplified and the original expression is better. This conversion is happening in `_eval_nseries` of `Expr` (L3024 in expr.py), where `collect(s1, x)` outputs the original expression but `collect(s1, x) + o` is outputting the changed expression. (where o is `O(_x**6)`)

I have tried a few things to make sympy output the desired expressions instead, but none of them worked and I am stuck. I would appreciate any help. If this is indeed an undesirable simplification, is it a bug not in the implementation but on the current master?

**Comment by 0sidharth** [comment]
I think some sort of distinction or a better alternative is required at - 
https://github.com/sympy/sympy/blob/cc946e09bdc27e7ab806e25a94f7beb9e0c0cd5d/sympy/core/expr.py#L2949-L2954 So that the Order terms of `1/x` do not simplify according to the logic of `x`

**Comment by jksuom** [comment]
> I think the answer sympy is finally returning is wrongly simplified and the original expression is better.

I agree. Asymptotic series are currently not handled in a satisfactory way. (The docstring of `Expr.aseries`even says that 'This is equivalent to `self.series(x, oo, n)`.' which is not true as asymptotic expansions can appear anywhere and expansions at `oo` can be convergent Laurent series.)

It seems to me that the series expansion code in `expr.py` needs to be improved but I have no detailed ideas about it. As a provisional fix I would try to apply this code https://github.com/sympy/sympy/blob/b6a1a3fede006b915e2cf61600abb4ba055df702/sympy/core/expr.py#L3029-L3032 
only if there `s1` has no (embedded) `Order` term.

File: sympy/core/expr.py [hunk 1]
```diff
@@ -3015,6 +3015,8 @@ def series(self, x=None, x0=0, n=6, dir="+", logx=None, cdir=0):
                 o = s1.getO()
                 s1 = s1.removeO()
             else:
+                if s1.has(Order):
+                    return s1
```
Comment (jksuom) [inline sympy/core/expr.py:3019]:
```suggestion
            elif s1.has(Order):
                # asymptotic expansion
                return s1
            else:
```
To bring forth the three cases:
* tailing order term
* embedded order term
* no order term

**Comment by jksuom** [comment]
Looks good, thanks! I have enabled auto-merge.

**Comment by 0sidharth** [comment]
Looks like a test from ode took a long time on the CI - 
```
sympy/solvers/ode/tests/test_single.py::test_factorable - Took 2742.024 seconds
```
However, I tried running this entire file again locally and it finished in the expected time - 
```
(sympy-dev) (error_aseries) ~/sympy $ bin/test sympy/solvers/ode/tests/test_single.py
==================================================================== test process starts =====================================================================
executable:         /home/sidharth/anaconda3/envs/sympy-dev/bin/python  (3.8.5-final-0) [CPython]
architecture:       64-bit
cache:              yes
ground types:       gmpy 2.0.8
numpy:              1.19.4
random seed:        23845405
hash randomization: on (PYTHONHASHSEED=1622638637)

sympy/solvers/ode/tests/test_single.py[34] ..w...w......w.ww..www............                                                                             [OK]

_______________________________________________________________________ slowest tests ________________________________________________________________________
sympy/solvers/ode/tests/test_single.py::test_Liouville_ODE - Took 11.154 seconds
sympy/solvers/ode/tests/test_single.py::test_nth_order_linear_euler_eq_nonhomogeneous_variation_of_parameters - Took 11.595 seconds
sympy/solvers/ode/tests/test_single.py::test_nth_linear_constant_coeff_var_of_parameters - Took 11.849 seconds
sympy/solvers/ode/tests/test_single.py::test_1st_exact - Took 12.669 seconds
sympy/solvers/ode/tests/test_single.py::test_nth_order_linear_euler_eq_nonhomogeneous_undetermined_coefficients - Took 13.157 seconds
sympy/solvers/ode/tests/test_single.py::test_nth_linear_constant_coeff_homogeneous - Took 16.183 seconds
sympy/solvers/ode/tests/test_single.py::test_2nd_linear_bessel_equation - Took 20.801 seconds
sympy/solvers/ode/tests/test_single.py::test_nth_algebraic - Took 22.992 seconds
sympy/solvers/ode/tests/test_single.py::test_separable_reduced - Took 28.637 seconds
sympy/solvers/ode/tests/test_single.py::test_2nd_2F1_hypergeometric - Took 46.420 seconds
sympy/solvers/ode/tests/test_single.py::test_lie_group - Took 49.506 seconds
sympy/solvers/ode/tests/test_single.py::test_separable - Took 67.436 seconds
sympy/solvers/ode/tests/test_single.py::test_2nd_nonlinear_autonomous_conserved - Took 84.460 seconds
sympy/solvers/ode/tests/test_single.py::test_factorable - Took 139.178 seconds
================================================== tests finished: 26 passed, 8 skipped, in 584.51 seconds ===================================================
```
I don't think it is an issue with the PR because as I said I tried it again (although master has not been merged), so is this a known thing? What could be the possible cause of this?

**Comment by jksuom** [comment]
I got 106 sec on master and 103 sec on this branch, which I think is normal variation. I cannot see any connection with this PR.

**Pull request merged**